pkgs/development/libraries/sqlcipher/default.nix

```
@@ -13,14 +13,15 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-E23PTNnVZbBQtHL0YjUwHNVUA76XS8rlARBOVvX6zZw=";
   };
 
-  nativeBuildInputs = [ installShellFiles ];
+  nativeBuildInputs = [ installShellFiles tcl ];
 
-  buildInputs = [ readline ncurses openssl tcl ];
+  buildInputs = [ readline ncurses openssl ];
 
   configureFlags = [ "--enable-threadsafe" "--disable-tcl" ];
 
   CFLAGS = [ "-DSQLITE_ENABLE_COLUMN_METADATA=1" "-DSQLITE_SECURE_DELETE=1" "-DSQLITE_ENABLE_UNLOCK_NOTIFY=1" "-DSQLITE_HAS_CODEC" ];
   LDFLAGS = lib.optional (readline != null) "-lncurses";
+  BUILD_CC = "${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}cc";
```

Wouldn't just

```suggestion
  BUILD_CC = "$CC_FOR_BUILD";
  
  depsBuildBuild = [ buildPackages.stdenv.cc ];
```

work? otherwise you can probably alias it in `preConfigure` or something.

---

I would prefer something like this as well, but CC_FOR_BUILD is not defined in the environment. I could not find a variable that contains the path of the correct compiler binary in a nix-shell.

---

```
[nix-shell:~/proj/nixpkgs]$ env | grep gcc
CC=x86_64-unknown-linux-musl-gcc
NIX_CC=/nix/store/svlxmzq6yacrgyp4caj9n1nng7srp8a7-x86_64-unknown-linux-musl-stage-final-gcc-wrapper-10.3.0
PATH=[...]

[nix-shell:~/proj/nixpkgs]$

---

Setting it to the musl cross-compiler would work in this case, since the produced binaries can run on the build machine as well, but what I currently have also works for `pkgsCross.aarch64-multiplatform.sqlcipher`.

---

Weird, this _should_ work as soon as you have `buildPackages.stdenv.cc` in `depsBuildBuild`:

```
nix-shell -E 'with import ./. {}; stdenv.mkDerivation { name = "foo"; depsBuildBuild = [ buildPackages.stdenv.cc ]; }'

[nix-shell:~/src/nix/nixpkgs]$ env | grep CC
CC=gcc
NIX_CC_FOR_BUILD=/nix/store/qvv5y4fx4x879rbsbs4g27mypl9wxbb9-gcc-wrapper-10.3.0
NIX_CC_WRAPPER_TARGET_HOST_x86_64_unknown_linux_gnu=1
NIX_CC=/nix/store/qvv5y4fx4x879rbsbs4g27mypl9wxbb9-gcc-wrapper-10.3.0
CC_FOR_BUILD=gcc
NIX_CC_WRAPPER_TARGET_BUILD_x86_64_unknown_linux_gnu=1
```

---

Oh, yes, with `depsBuildBuild = [ buildPackages.stdenv.cc ];` it works. Thanks for the improvement 🙂